fastparse/shared/src/main/scala/fastparse/Api.scala

```diff
@@ -12,6 +12,8 @@ trait Api{
 
   val Result = core.Result
   type Result[+T] = core.Result[T]
+  val Failure = core.Result.Failure
+  val Success = core.Result.Success
```

Comment:
I somewhat-intentionally left these out, because I Want them to be called via `Result.*` rather than just `*`. Otherwise they tend to collide with lots of things called `Success` or `Failure`, e.g. `Try` in the standard library =/ We could name them `ParseSuccess` or `ParseFailure` but I'm more inclined to just name them `Result.Success` and `Result.Failure`